src/Neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -54,16 +54,29 @@ public void AddTransaction(Transaction tx)
         /// <param name="tx">The specified <see cref="Transaction"/>.</param>
         /// <param name="conflictingTxs">The list of <see cref="Transaction"/> that conflicts with the specified one and are to be removed from the pool.</param>
         /// <param name="snapshot">The snapshot used to verify the <see cref="Transaction"/>.</param>
+        /// <param name="settings">The <see cref="ProtocolSettings"/> used to verify the <see cref="Transaction"/>.</param>
         /// <returns><see langword="true"/> if the <see cref="Transaction"/> passes the check; otherwise, <see langword="false"/>.</returns>
-        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot)
+        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot, ProtocolSettings settings)
         {
-            var balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
+            BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             _senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
 
             var expectedFee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
             foreach (var conflictTx in conflictingTxs.Where(c => c.Sender.Equals(tx.Sender)))
                 expectedFee -= conflictTx.NetworkFee + conflictTx.SystemFee;
-            if (balance < expectedFee) return false;
+            if (balance < expectedFee)
+            {
+                // Only consider unclaimed GAS when claimed balance is insufficient and HF_Faun is active.
+                if (settings.Hardforks.TryGetValue(Hardfork.HF_Faun, out var faunHeight) &&
```

Comment:
@roman-khimov  Maybe we can have a special attribute to "mark" this tx as this. "AllowUseClaimGasAttribute"

Reply:
Neat idea if this attribute costs something. Then it can autoclaim irrespective of the current balance. But then it will be used for autoclaiming "along the way" (while sending other transactions), so the cost should be comparable to NEO flip or vote (IIRC voting is cheaper than transfer for GAS claims).